tests/rbd/rbd_abort_image_migration.py

```diff
@@ -0,0 +1,128 @@
+"""Module to execute abort Live migration feature test case.
+
+Pre-requisites :
+We need cluster configured with atleast one client node with ceph-common package,
+conf and keyring files
+
+Test cases covered :
+1)CEPH-83573324 - Copy the image from source to destination and stop in between
+and check the appropriate message
+
+Test Case Flow :
+1. Create EC pool with default k_m values(2,2) and an image
+2. Create an another EC pool with default k_m values(2,2) for destination pool
+3. Mount the images and run io from clients
+4. Stop IO and prepare migration process from EC pool to EC pool
+5. Start IO from client on destination images
+6. Execute migration
+7. Abort migration and verify that cross-links from source to destination is removed,
+and remove the target image.
+"""
+
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import (
+    Rbd,
+    initial_rbd_config,
+    verify_migration_abort,
+    verify_migration_state,
+)
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def run(**kw):
+    """verify Abort image migration from EC pool to EC pool.
+
+    Args:
+        kw: Key/value pairs of configuration information to be used in the test
+            Example::
+            config:
+                "ec-pool-only": True
+                ec_pool_config:
+                    pool: rbd_pool_4
+                    data_pool: rbd_ec_pool_4
+                    ec_profile: rbd_ec_profile_4
+                    image: rbd_image_4
+                    size: 10G
+
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+    """
+    log.info("Starting execution for abort image migration scenario...")
+
+    for key, value in kw["config"].items():
+        if key == "source":
+            kw["config"].update(value)
+            initial_rbd_config(**kw)
+            for key in kw["config"]["source"]:
+                kw["config"].pop(key)
+        elif key == "destination":
+            kw["config"].update(value)
+            initial_rbd_config(**kw)
+            for key in kw["config"]["destination"]:
+                kw["config"].pop(key)
+
+    rbd = Rbd(**kw)
+    try:
+        client = kw["ceph_cluster"].get_nodes(role="client")[0]
+
+        if kw.get("config").get("source").get("ec_pool_config"):
+            src_pool = kw["config"]["source"]["ec_pool_config"]["pool"]
+            image = kw["config"]["source"]["ec_pool_config"]["image"]
+            src_spec = src_pool + "/" + image
+            dest_pool = kw["config"]["destination"]["ec_pool_config"]["pool"]
+            dest_spec = dest_pool + "/" + image
+        else:
+            src_pool = kw["config"]["source"]["rep_pool_config"]["pool"]
```

Comment:
As this test is running for `ec-pool-only: True` then why rep_pool_config needed in else part?

Reply:
Done.